python/datasources/phrma.py

```diff
@@ -0,0 +1,354 @@
+            f'{std_col.STATINS_PREFIX}_{COUNT_NO}',
+            f'{std_col.BETA_BLOCKERS_PREFIX}_{COUNT_NO}',
+            f'{std_col.RASA_PREFIX}_{COUNT_NO}',
+            f'{std_col.STATINS_PREFIX}_{ADHERENCE_RATE}',
+            f'{std_col.BETA_BLOCKERS_PREFIX}_{ADHERENCE_RATE}',
+            f'{std_col.RASA_PREFIX}_{ADHERENCE_RATE}'
+        ])
+
+        if demo_breakdown == std_col.RACE_OR_HISPANIC_COL:
+            std_col.add_race_columns_from_category_id(df)
+
+        df = df.sort_values(
+            by=[fips_to_use, demo_col]).reset_index(drop=True)
+
+        return df
+
+
+def load_phrma_df_from_data_dir(geo_level: str, breakdown: str) -> pd.DataFrame:
+    """ Generates Phrma data by breakdown and geo_level
+    geo_level: string equal to `county`, `national`, or `state`
+    breakdown: string equal to `age`, `race_and_ethnicity`, `sex`, or `all`
+    return: a single data frame of data by demographic breakdown and
+        geo_level with data columns loaded from multiple Phrma source tables """
+
+    sheet_name = get_sheet_name(geo_level, breakdown)
+
+    merge_cols = [std_col.STATE_FIPS_COL,
+                  std_col.STATE_NAME_COL]
+    if geo_level == COUNTY_LEVEL:
+        merge_cols.extend([std_col.COUNTY_FIPS_COL, std_col.COUNTY_NAME_COL])
+
+    if breakdown != TMP_ALL:
+        breakdown_col = std_col.RACE_CATEGORY_ID_COL if breakdown == std_col.RACE_OR_HISPANIC_COL else breakdown
+        merge_cols.append(breakdown_col)
+    fips_col = std_col.COUNTY_FIPS_COL if geo_level == COUNTY_LEVEL else std_col. STATE_FIPS_COL
+    fips_length = 5 if geo_level == COUNTY_LEVEL else 2
+
+    topic_dfs = []
+
+    for condition in PHRMA_CONDITIONS:
+
+        topic_df = gcs_to_bq_util.load_csv_as_df_from_data_dir(
+            PHRMA_DIR,
+            f'{condition}-{sheet_name}.csv',
+            subdirectory=condition,
+            dtype=DTYPE,
+            na_values=["."],
+        )
+
+        if geo_level == NATIONAL_LEVEL:
+            topic_df[STATE_CODE] = US_FIPS
+            topic_df[STATE_NAME] = US_NAME
+
+        topic_df = rename_cols(topic_df,
+                               cast(GEO_TYPE, geo_level),
+                               cast(SEX_RACE_ETH_AGE_TYPE, breakdown),
+                               condition)
+
+        topic_dfs.append(topic_df)
+
+    df_merged = reduce(lambda df_a, df_b: pd.merge(df_a, df_b, on=merge_cols,
+                                                   how='outer'), topic_dfs)
+
+    # drop rows that dont include FIPS and DEMO values
+    df_merged = df_merged[df_merged[fips_col].notna()]
+    df_merged = ensure_leading_zeros(df_merged, fips_col, fips_length)
+
+    return df_merged
+
+
+def get_sheet_name(geo_level: str, breakdown: str) -> str:
+    """ geo_level: string equal to `county`, `national`, or `state`
+   breakdown: string equal to `age`, `race_and_ethnicity`, `sex`, or `all`
+   return: a string sheet name based on the provided args  """
+
+    sheet_map = {
```

Comment:
I like that you're defining the map inside the function rather than declaring it at the top of the file. It makes it so much easier to read.